pkg/cidata/cidata.TEMPLATE.d/lima.env

```diff
@@ -15,3 +15,6 @@ LIMA_CIDATA_CONTAINERD_SYSTEM=1
 LIMA_CIDATA_CONTAINERD_SYSTEM=
 {{- end}}
 LIMA_CIDATA_SLIRP_GATEWAY={{ .SlirpGateway }}
+{{- range $key, $val := .EnvironmentVariables}}
```

Comment:
This isn't going to work; the `lima.env` file is only used during `cloud-init`; you still need to copy them into a profile script, e.g. `/etc/profile.d/lima.sh`.

In `lima.env` the variables should be defined like this (similar to how `.Mounts` are defined earlier):

```shell
LIMA_CIDATA_ENVS=2
LIMA_CIDATA_ENVS_0_NAME=http_proxy
LIMA_CIDATA_ENVS_0_VALUE=my.proxy.com:8080
LIMA_CIDATA_ENVS_1_NAME=https_proxy
LIMA_CIDATA_ENVS_1_VALUE=my.proxy.com:8080
```

and then in `pkg/cidata/boot/25-guestagent-base.sh` you need a loop that writes them to `/etc/profile.d/lima.sh`.

Reply:
> /etc/profile.d/lima.sh

Should the path be `/etc/environment`?

Reply:
> Should the path be `/etc/environment`?

I don't think it really matters. The only advantage I see for a profile script is that you can extend `PATH`:

```
env:
    PATH="/opt/bin:$PATH"
```

But you can always do this with a provisioning script as well.

Another thought: `/etc/environment` is not a POSIX feature but is implemented by PAM. So not sure if that always works. I guess I should try the minimal Alpine image...

Reply:
So I was thinking more during the cloud-init phase. In my local testing I added them directly to lima.env and was about resolve the issues I was having during the apt-get update and apt-get install. Anything after that I figured if a user needed them they could add anything they need during the provision stage. 

But can add the additional logic so they can get added to the profile.

Reply:
> So I was thinking more during the cloud-init phase.

Ah yes, thank you! I forgot that these also need to be in effect during cloud-init.

This could be done by sourcing `/etc/profile.d/lima.sh` in `pkg/cidata/boot/30-install-packages.sh`.

Reply:
> But can add the additional logic so they can get added to the profile.

I would prefer this to be a global conf setting and not just for the cloud-init phase.

But wait for @AkihiroSuda to decide on `/etc/environment` vs. `/etc/profile.d/lima.sh`

Reply:
Either is fine, as long as we can have CI. The latter seems more friendly to Alpine.

Reply:
Yes, alpine-virt ignores `/etc/environment` because it doesn't use PAM by default.

`limactl shell` has a hardcoded `exec bash --login`, so using `/etc/profile.d` seems appropriate.